python/dgl/distributed/dist_graph.py

```diff
@@ -528,34 +534,33 @@ def __init__(self, graph_name, gpb=None, part_config=None):
             self._num_nodes += int(part_md['num_nodes'])
             self._num_edges += int(part_md['num_edges'])
 
+        # Get canonical edge types.
+        if not self._gpb.init_by_canonical_etypes:
+            # Deprecated since 0.9.1 and for backward compatible only.
+            # This requires the server to store the graph with coo format.
+            eid = []
+            for etype in self.etypes:
+                type_eid = F.zeros((1,), F.int64, F.cpu())
+                eid.append(self._gpb.map_to_homo_eid(type_eid, etype))
+            eid = F.cat(eid, 0)
+            src, dst = dist_find_edges(self, eid)
+            src_tids, _ = self._gpb.map_to_per_ntype(src)
+            dst_tids, _ = self._gpb.map_to_per_ntype(dst)
+            self._canonical_etypes = [None] * len(self.etypes)
```

Comment:
No need to have `self._canonical_etypes`. It can be local variable. The class can have a property `canonical_etypes` that returns `self._gdb._canonical_etypes`.